sklearn/decomposition/tests/test_nmf.py

```diff
@@ -589,12 +639,78 @@ def test_nmf_custom_init_dtype_error():
     W = rng.random_sample((20, 15))
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
-        NMF(init='custom').fit(X, H=H, W=W)
+        Estimator(init='custom').fit(X, H=H, W=W)
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
         non_negative_factorization(X, H=H, update_H=False)
 
 
+def test_nmf_close_minibatch_nmf():
+    # Test that the decomposition with standard and minibatch nmf
+    # gives close results
+    rng = np.random.mtrand.RandomState(42)
+    X = np.abs(rng.randn(48, 5))
+    nmf = NMF(5, solver='mu', init='nndsvdar', random_state=0,
+              max_iter=2000, beta_loss='kullback-leibler')
+    mbnmf = MiniBatchNMF(5, solver='mu', init='nndsvdar', random_state=0,
+                         max_iter=2000, beta_loss='kullback-leibler',
+                         batch_size=48)
```

Comment:
I tried changing this test with `batch_size=24`, and it does not work at all, even with reduced precision.
It seems that the MiniBatchNMF results in a different scaling of W versus H, so to be agnostic to different scalings, I tried to compare the reconstruction errors `W @ H - X`. But even with that, it looks like the minibatch implementation does not reach errors as low as the batch implementation. There seems to be something wrong here. Could it be related to the comment `# TODO internal iters for W` in `MiniBatchNMF.fit_transform` ?
![Figure_1](https://user-images.githubusercontent.com/11065596/103028067-decf0c80-450b-11eb-8cd3-42114ca43dba.png)
```py
    rng = np.random.mtrand.RandomState(42)
    X = np.abs(rng.randn(48, 5))

    results = []
    max_iter_list = np.int_(np.logspace(0, 2, 30))
    for max_iter in max_iter_list:
        nmf = NMF(5, solver='mu', init='nndsvdar', random_state=0,
                  max_iter=max_iter, beta_loss=2)
        mbnmf = MiniBatchNMF(5, solver='mu', init='nndsvdar', random_state=0,
                             max_iter=max_iter, beta_loss=2, batch_size=24)
        nmf.fit(X)
        mbnmf.fit(X)
        results.append([nmf.reconstruction_err_,
                        mbnmf.reconstruction_err_])
    import matplotlib.pyplot as plt
    plt.plot(max_iter_list, np.array(results), '.-')
    plt.legend(['NMF', 'MiniBatchNMF'])
    plt.xlabel('iterations')
    plt.ylabel('Reconstruction error')
    plt.show()
```

Reply:
Thanks @TomDLT , I've parametrized the wrong test... :( ... I see what you mean now. I will have a deep look at it after the Christmas break.